### PR TITLE
Disallow reporting from known VPN/proxy addresses.

### PR DESCRIPTION
**Pull request by glennguy**
Also disallow user submitted reports from overseas and known VPN/proxy
addresses

### CODE REVIEW

File: lib/aussieaddonscommon/utils.py [hunk 1]
```diff
@@ -218,18 +218,49 @@ def log_kodi_platform_version():
...
+
+    if not valid_country:
+        country_code = connection_info.get('country')
+        if country_code:
+            import countries
+            country_name = countries.countries.get(country_code, country_code)
+            message.append('Your country is reported as %s, but this service '
+                           'is probably geo-blocked to Australia.' %
+                           country_name)
+            xbmcgui.Dialog().ok(*message)
```
Comment (andybotting) [inline lib/aussieaddonscommon/utils.py:237]: how about putting a 'return' right under here instead

File: lib/aussieaddonscommon/utils.py [hunk 2]
```diff
@@ -218,18 +218,49 @@ def log_kodi_platform_version():
     log("Kodi %s running on %s" % (version, platform))
 
 
+
+def check_country(connection_info, message=None):
```
Comment (andybotting) [inline lib/aussieaddonscommon/utils.py]: this function returns a bool, so would be nice to have a clear method name here like is_valid_country or valid_country?

File: lib/aussieaddonscommon/utils.py [hunk 3]
```diff
@@ -218,18 +218,49 @@ def log_kodi_platform_version():
...
+                           'is probably geo-blocked to Australia.' %
+                           country_name)
+            xbmcgui.Dialog().ok(*message)
+    
+    if blacklisted_hostname:
+        message.append('VPN/proxy detected that has been blocked by this '
+                       'content provider.')
+        xbmcgui.Dialog().ok(*message)
+
+    return valid_country and not blacklisted_hostname
```
Comment (andybotting) [inline lib/aussieaddonscommon/utils.py]: maybe say return True here and add a return False after the blacklisted_hostname check?

**Review by andybotting — CHANGES_REQUESTED** [review]
Looks good, with a few comments.

**Comment by glennguy** [comment]
Thanks Andy, changes made :)

**Review by andybotting — APPROVED** [review]

**Pull request merged**